From 2be4413c064cdbf944c7d58fef205af461874520 Mon Sep 17 00:00:00 2001
From: ZhangPeng <zhangpeng362@huawei.com>
Date: Wed, 30 Nov 2022 08:28:32 +0000
Subject: [PATCH 0113/1398] [Huawei] kfence: add the cmdline params of the
 kfence_xxx type

Offering: HULK
hulk inclusion
category: feature
bugzilla: 188062

-------------------------------

Currently, the command line parameter "xxx.xxx" with a dot in the
cmdline is not supported due to a service bug. Add equivalent parameters
without a dot in the name. And these parameters support all
architectures.

Signed-off-by: ZhangPeng <zhangpeng362@huawei.com>
---
 mm/kfence/core.c | 20 ++++++++++++++++----
 1 file changed, 16 insertions(+), 4 deletions(-)

diff --git a/mm/kfence/core.c b/mm/kfence/core.c
index f67418a30282..7ca98dac84c1 100644
--- a/mm/kfence/core.c
+++ b/mm/kfence/core.c
@@ -105,7 +105,6 @@ static const struct kernel_param_ops sample_interval_param_ops = {
 };
 module_param_cb(sample_interval, &sample_interval_param_ops, &kfence_sample_interval, 0600);
 
-#ifdef CONFIG_ARM64
 static int __init parse_sample_interval(char *str)
 {
 	long num;
@@ -126,7 +125,14 @@ static int __init parse_sample_interval(char *str)
 	kfence_sample_interval = (unsigned long)num;
 	return 0;
 }
-early_param("kfence.sample_interval", parse_sample_interval);
+early_param("kfence_sample_interval", parse_sample_interval);
+
+#ifdef CONFIG_ARM64
+static int __init arm64_parse_sample_interval(char *str)
+{
+	return parse_sample_interval(str);
+}
+early_param("kfence.sample_interval", arm64_parse_sample_interval);
 #endif
 
 /* Pool usage% threshold when currently covered allocations are skipped. */
@@ -187,7 +193,6 @@ static const struct kernel_param_ops num_objects_param_ops = {
 };
 module_param_cb(num_objects, &num_objects_param_ops, &kfence_num_objects, 0600);
 
-#ifdef CONFIG_ARM64
 static int __init parse_num_objects(char *str)
 {
 	unsigned long num;
@@ -199,7 +204,14 @@ static int __init parse_num_objects(char *str)
 	kfence_num_objects = num;
 	return 0;
 }
-early_param("kfence.num_objects", parse_num_objects);
+early_param("kfence_num_objects", parse_num_objects);
+
+#ifdef CONFIG_ARM64
+static int __init arm64_parse_num_objects(char *str)
+{
+	return parse_num_objects(str);
+}
+early_param("kfence.num_objects", arm64_parse_num_objects);
 #endif
 #endif
 
-- 
2.17.1

